tests/components/test_grid_component.py

```diff
@@ -106,19 +112,19 @@ def test_create_grid_from_bbox(mock_model):
     grid_component.create(
         region={"bbox": bbox},
         res=0.05,
-        add_mask=False,
+        add_mask=True,
```

Comment:
It was unclear from the test if `add_mask` was part of the spec of this test, but otherwise I would retrieve a GeoDataFrame without data and couldn't check the region.

Reply:
But the case of add_mask is False should still be handled and should return a return (in that case just raster.box) Which is why I doubt the set_region method and region should really just be derived from the final component geometry.

Reply:
Setting `add_mask=False` gives me a `self.grid` with:

```
<xarray.Dataset> Size: 188B
Dimensions:      (y: 7, x: 16)
Coordinates:
  * y            (y) float64 56B 45.63 45.58 45.53 45.48 45.43 45.38 45.33
  * x            (x) float64 128B 12.08 12.13 12.18 12.23 ... 12.73 12.78 12.83
    spatial_ref  int32 4B 0
Data variables:
    *empty*
```

The functions for `bounds` and `region` in GridComponent now check for `if len(self.data) > 0:`.
This might actually be removed? `self.data.raster.bounds` does contain a value: `(12.05, 45.3, 12.850000000000001, 45.650000000000006)`

Reply:
Yes so we need better checks on the grid property in case mask is False indeed

Reply:
For me it is unclear what I should change now

Reply:
Should we check for `if self.data.raster is not None`, or `if len(self.data.coords)>0`?

Reply:
@hboisgon This is still a thing after all the refactors. The code in `GridComponent._region_data` still checks for `if len(self.data) > 0:`. What should we make of this?

Reply:
I think we decided to leave it for now as it did not work for main already and open an issue to fix it for later as it may not be fully straightforward to fix